### PR TITLE
Implement bring forward functionality

### PR DESCRIPTION
**Pull request by kyle1morel**
# Description

This is the completed implementation of the bring forward functionality. It is a rebase onto master of all work/merges to this feature branch. All code in this PR has been reviewed and approved within the following PRs that were merged to this feature branch.

https://github.com/bcgov/nr-permitconnect-navigator-service/pull/55
https://github.com/bcgov/nr-permitconnect-navigator-service/pull/49

3c716ba Package & Node version updates
cd53be2 Add optional bring forward state filter
615208b Resolve various testing issues
34c4ee1 Resolve rebase conflicts
ff45a59 Update Calendar to work with PrimeVue 3.51.0
7c77e04 Add colour coded bring forward notifications to submission list view
47be851 Implement Bring Forward tab & table
a0d7321 Bring forward feature migration, unit tests, and validation
a8ae554 Note model and functionality changes
0fd3e1d Bring forward frontend features

## Types of changes

New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/src/services/note.ts [hunk 1]
```diff
@@ -24,11 +24,29 @@ const service = {
     return note.fromPrismaModel(response);
   },
 
+  /**
+   * @function listBringForward
+   * Retrieve a list of notes with the Bring forward type
+   * @returns {Promise<Note[]>} The result of running the findMany operation
+   */
+  listBringForward: async () => {
```
[the remark this answers is not in the document]
Comment (TimCsaky) [reply app/src/services/note.ts:32]: actually you can ignore this comment.

File: app/src/db/models/note.ts [hunk 2]
```diff
@@ -14,6 +14,8 @@ export default {
     return {
       note_id: input.noteId as string,
       activity_id: input.activityId,
+      bring_forward_date: input.bringForwardDate ? new Date(input.bringForwardDate) : null,
+      bring_forward_state: input.bringForwardState,
```
[the remark this answers is not in the document]
Comment (kyle1morel) [reply app/src/db/models/note.ts:18]: Both sides are `string | null`, so it's okay as is

File: frontend/src/components/submission/SubmissionBringForwardCalendar.vue [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+          :sortable="true"
+        />
+        <Column
+          field="createdByFullName"
+          header="Navigator"
+          :sortable="true"
+        />
+        <Column
+          field="bringForwardDate"
+          header="Date"
```
[the remark this answers is not in the document]
Comment (kyle1morel) [reply frontend/src/components/submission/SubmissionBringForwardCalendar.vue:89]: I agree, but the design just says `Date`, so leaving as is for now.

File: frontend/src/views/SubmissionsView.vue [hunk 4]
```diff
@@ -48,5 +138,22 @@ onMounted(async () => {
         <span v-else>Failed to load statistics.</span>
       </div>
     </TabPanel>
+    <TabPanel header="Bring Forward Calendar">
+      <SubmissionBringForwardCalendar :bring-forward="bringForward" />
+    </TabPanel>
   </TabView>
 </template>
+
+<style scoped lang="scss">
```
[the remark this answers is not in the document]
Comment (kyle1morel) [reply frontend/src/views/SubmissionsView.vue:148]: Brought up with Tyler.

**Pull request merged**